build/bootstrap.sh

```diff
@@ -0,0 +1,220 @@
+    # Additional Boost libs for cpprestsdk
+    apt-get install -y libboost-random-dev libboost-regex-dev
+    apt-get install -y libproxy-dev libssl-dev uuid-dev
+
+    # Install cpprest dependencies
+    # libssl-dev also required but installed above because plugin uses libssl-dev directly
+    apt-get install -y zlib1g-dev
+
+    # Most target platforms do not natively have a version of cpprest that supports url-redirection
+    # Build and install v2.10.16 as it's the earliest version which supports url-redirection
+    rm -rf /tmp/cpprestsdk
+    mkdir /tmp/cpprestsdk
+    cd /tmp/cpprestsdk
+    git clone https://github.com/microsoft/cpprestsdk.git .
+    git checkout tags/v2.10.16
+    git submodule update --init
+    mkdir /tmp/cpprestsdk/build
+    cd /tmp/cpprestsdk/build
+    cmake -G Ninja -DCMAKE_BUILD_TYPE=minsizerel -DCMAKE_POSITION_INDEPENDENT_CODE=ON -DBUILD_SHARED_LIBS=OFF -DBUILD_TESTS=OFF -DBUILD_SAMPLES=OFF -Wno-dev -DWERROR=OFF ..
+    ninja
+    ninja install
+
+    rm -rf /tmp/gtest
+    mkdir /tmp/gtest
+    cd /tmp/gtest
+
+    if [[ "$PLATFORM" == "ubuntu2004" || "$PLATFORM" == "debian10" ]];
+    then
+        # The latest native-version of gtest on ubuntu2004 and debian10 currently has a bug where CMakeLists doesn't declare an install target, causing 'make install' to fail
+        # Clone from source and use release-1.10.0 instead, since gtest is a source package anyways
+        git clone https://github.com/google/googletest.git .
+        git checkout release-1.10.0
+        mkdir cmake 
+        cd cmake
+        cmake /tmp/gtest
+        make 
+        make install 
+    else
+        # libgtest-dev is a source package and requires manual installation
+        apt-get -y install libgtest-dev
+        cmake /usr/src/gtest
+        make
+        make install
+    fi
+}
+
+function installDeveloperTools
+{
+    echo "[INFO] Installing developer tools"
+    apt install -y python-pip
+    pip install cpplint
+
+    # Installs to a non-standard location so add to PATH manually
+    export PATH=$PATH:~/.local/bin
+}
+
+# TODO(jimson): Running docker builds on 1ES hosted agents are currently blocked due to az pipeline agent not having permissions to call the docker daemon
+# Tried potential solutions here: https://docs.docker.com/engine/security/rootless/, to no avail.
+# 1. Using 'usermod -aG docker $USER' hangs the az pipeline agent on 'newgrp docker', which is required for the permission changes to take effect
+# 2. The rootless install succeeds, but azdo pipeline agent still reports permissions failure when calling into docker daemon
+# See if we can leverage someone elses docker install task from Image Factory when provisioning the 1ES managed image
+# After docker is callable on a 1ES managed image, we can swap all pipelines to use 1ES hosted pool instead of the microsoft hosted agent, and we can remove usage of the bootstrap script
```

Comment:
This comment can now be deleted

Reply:
The lines above, about permissions issue calling into docker daemon, can be removed also, right?

How did you resolve it?

Reply:
Yes, I added comment on codeflow - but I think on github it only highlights the bottom-most line. The whole comment block can be removed.

So this comment came out of my first PR for this work, where I was using an Ubuntu1804 base image with no dependencies installed. The issue was that when trying to install docker via pipeline step - the current user is running as the self-hosted agent, with network service permissions. As a result - the pipeline agent would fail to perform certain commands, like pulling from a remote container registry.
When running this script as an ImageFactory artifact - docker gets installed with the right permissions to have the docker CLI callable through the pipeline agent's user. 

So the tl;dr is that it self-resolved once I did the install as a part of image provisioning rather than trying to install docker as a pipeline step.